core/src/main/java/org/elasticsearch/ResourceNotFoundException.java

```diff
@@ -0,0 +1,52 @@
+package org.elasticsearch;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.rest.RestStatus;
+
+import java.io.IOException;
+
+/**
+ * Generic ResourceNotFoundException corresponding to the {@link RestStatus#NOT_FOUND} status code
+ */
+public class ResourceNotFoundException extends ElasticsearchException {
```

Comment:
maybe we should also have a a `es.resource.id` header and have a method to add it - `addResourceId` (a la `addIndex`) ?